core/GlobalState.cc

```diff
@@ -1412,6 +1418,72 @@ bool GlobalState::wasModified() const {
     return wasModified_;
 }
 
+bool GlobalState::isTypecheckingCanceled() const {
+    return lspEpochInvalidator->load() != currentlyProcessingLSPEpoch->load();
+}
+
+void GlobalState::startCommitEpoch(u4 epoch) {
+    absl::MutexLock lock(epochMutex.get());
+    currentlyProcessingLSPEpoch->store(epoch);
+    lspEpochInvalidator->store(epoch);
+    // These should always be different.
+    ENFORCE(epoch != lastCommittedLSPEpoch->load());
+}
+
+optional<pair<u4, u4>> GlobalState::getRunningSlowPath() const {
```

Comment:
This will always return `nullopt` when cancelation is disabled, as these two epochs will have the value `0`.

Reply:
please leave this as a comment in source

Reply:
This is no longer true now that multithreading isn't special. It always returns the running slow path, or nullopt if a slow path isn't running. I've amended the docstring in `GlobalState.h`.